manifests/params.pp

```diff
@@ -168,19 +168,21 @@
   $auth_allowed = ['$1']
 
   # Will this host be a puppet agent ?
-  $agent                     = true
-  $remove_lock               = true
-  $client_certname           = $::clientcert
+  $agent                      = true
+  $remove_lock                = true
+  $client_certname            = $::clientcert
 
   # Custom puppetmaster
-  if defined('$trusted') and $::trusted['authenticated'] == 'local' {
-    $puppetmaster            = undef
+  # needed due to a PUP-4072
+  # more information in https://github.com/theforeman/puppet-foreman/commit/5fe3239da0c6fbac76172f61042a69ab3a7eb4e6
+  if versioncmp($::puppetversion, '3.7.5') < 0 or defined('$::puppetmaster') {
```

Comment:
It was needed for masterless setups with strict variables enabled, but this should have it work in all strict variable setups (from 3.7.5 onwards). I think we preferred this over making it 3.6-only at the time in #237.

In non-strict either masterless or master setups then it should take the `else` branch and leave the parameter set to undef.